### PR TITLE
GPU-aware MPI additions for Frontier

### PR DESCRIPTION
**Pull request by jrood-nrel**
I'm not sure if we want these since it makes is slower, but since I put it all together it might be worth getting it into a mergable state.

### CODE REVIEW

File: repos/exawind/packages/amr-wind/package.py [hunk 1]
```diff
@@ -41,6 +43,21 @@ def setup_build_environment(self, env):
             env.set("LSAN_OPTIONS", "suppressions={0}".format(join_path(self.package_dir, "sup.asan")))
         if "%intel" in self.spec:
             env.append_flags("CXXFLAGS", "-no-ipo")
+        if "+mpi" in self.spec and "+gpu-aware-mpi" in self.spec and "+rocm" in self.spec and find_machine(verbose=False, full_machine_name=False) == "frontier":
```
Comment (psakievich) [inline repos/exawind/packages/amr-wind/package.py]: Minor nitpick, but if we `require('+mpi', when='+gpu-aware-mpi')` then we don't need to check both.

**Review by psakievich — APPROVED** [review]

**Comment by psakievich** [comment]
I think we should have this, but default it to off.

File: repos/exawind/packages/amr-wind/package.py (hunk 1 above)
Comment (jrood-nrel) [reply repos/exawind/packages/amr-wind/package.py]: I agree.

**Comment by jrood-nrel** [comment]
For posterity, I should also link to the documentation I was following for this: https://docs.olcf.ornl.gov/systems/frontier_user_guide.html#gpu-aware-mpi

File: repos/exawind/packages/amr-wind/package.py (hunk 1 above)
Comment (psakievich) [reply repos/exawind/packages/amr-wind/package.py]: we should also add `conflicts("+gpu-aware-mpi", when="~cuda~rocm")`

Comment (psakievich) [reply repos/exawind/packages/amr-wind/package.py]: kind of pedantic but would be good for making sure the concretizer doesn't screw us over with something weird for debug cpu only builds on gpu machines

**Comment by marchdf** [comment]
@jrood-nrel and @psakievich can we merge this? I am actually using it ;)

**Comment by psakievich** [comment]
@marchdf I made a few changes based on my previous comments. Want to test them out for me really quick? 😉  Looks like something is up with CI too, but I don't think that is related to my changes. I'll investigate.

**Comment by marchdf** [comment]
Yes I can test that now!

File: repos/exawind/packages/amr-wind/package.py [hunk 2]
```diff
@@ -35,13 +37,32 @@ class AmrWind(SMCMakeExtension, bAmrWind):
     depends_on("hypre+umpire", when="+umpire")
     depends_on("hypre+sycl", when="+sycl")
     depends_on("hypre+gpu-aware-mpi", when="+gpu-aware-mpi")
+    depends_on("+mpi", when="+gpu-aware-mpi")
```
Comment (marchdf) [inline repos/exawind/packages/amr-wind/package.py]: `+mpi` should be `mpi`

**Comment by marchdf** [comment]
with the change I suggest above, this builds on frontier.

**Pull request merged**